src/components/ExchangeIssuance/components/IssueRedeemButton.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+
+import { RoundedButton } from 'components/RoundedButton'
+import useExchangeIssuance from 'hooks/useExchangeIssuance'
+import useWallet from 'hooks/useWallet'
+import useApproval from 'hooks/useApproval'
+import {
+  daiTokenAddress,
+  usdcTokenAddress,
+  dpiTokenAddress,
+  cgiTokenAddress,
+  indexTokenAddress,
+  uniswapRouterAddress,
+} from 'constants/ethContractAddresses'
+
+/**
+ * IssueRedeemButton - Displays a button used in the issue redeem flow.
+ * The button can be used to:
+ * 1. Prompt user login to complete a transaction
+ * 2. Prompt the user to approve input currency to Uniswap Router
+ * 3. Execute the trade transaction
+ */
+const IssueRedeemButton: React.FC = () => {
+  const {
+    issuanceToken,
+    isFetchingOrderData,
+    isUserIssuing,
+    currencyQuantity,
+    tokenQuantity,
+    issuanceData,
+    selectedCurrency,
+    onExecuteIssuance,
+  } = useExchangeIssuance()
+
+  const { account, onOpenWalletModal } = useWallet()
+  const daiApproval = useApproval(daiTokenAddress, uniswapRouterAddress)
+  const usdcApproval = useApproval(usdcTokenAddress, uniswapRouterAddress)
+  const dpiApproval = useApproval(dpiTokenAddress, uniswapRouterAddress)
+  const cgiApproval = useApproval(cgiTokenAddress, uniswapRouterAddress)
+  const indexApproval = useApproval(indexTokenAddress, uniswapRouterAddress)
```

Comment:
approvals should be to exchange issuance contract address

Reply:
Yes fixed